docs/developer/contributing.rst

```diff
@@ -204,8 +204,8 @@ To install ``nbstripout``, run the following within the pymedphys directory:
 
 .. code:: bash
 
-    λ conda create --name pmp python=3.7 # Optional.
-    λ conda activate pmp # Optional
+    λ conda create --name pmp python=3.7 shapely
```

Comment:
```suggestion
    λ conda create --name pmp python=3.7 shapely nbstripout
```

Reply:
Good idea! Although I'm still thinking `nbstripout` is best installed in base...

Reply:
Oh that's fine. Installed in base only? Does that work? ...actually K think it might, because nbstripout install directly pins to the executable, not the path call. Yeah, just change that here.

Deal with the environment part in a different section of the docs, have the environment happen right before the `yarn bootstrap:dev:all` and maybe if run a `conda deactivate` before nbstripout install, just to be sure...

Reply:
...to make that work we'll need to remove nbstripout from the bootstrap code. Otherwise it will unknowingly replace the base version with an nbstripout installed within the env.

Reply:
Hmm ... Further thought... That might not work ... Anyway, test it out.

Reply:
We could make the hook try and specifically call the nbstripout that is installed? As in, instead of it calling the nbstripout that is within the PATH have it call the one that is referenced in the git nbstripout installation.

That should fix all the issues, that should make it work in VSCode regardless...

Reply:
Installing `nbstripout` in `base` means it is always in `%PATH%`. So it doesn't matter whether you have an env activated or not, nor does it matter if `nbstripout` is not installed within an env that you have activated. In both cases, `nbstripout` can still be found in %PATH% and so should still work, I believe. Moreover, other apps that don't have that env activated will still see `nbstripout` and it also protects the case where people are accidentally committing to `pymedphys` without their env activated (not the hardest mistake to make).

Observe:
```bash
λ conda list | grep nbstripout
nbstripout                0.3.5                      py_0    conda-forge
λ conda create --name no-nbstripout python=3.7
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\Centr\Anaconda3\envs\no-nbstripout

  added / updated specs:
    - python=3.7


The following NEW packages will be INSTALLED:

  ca-certificates    conda-forge/win-64::ca-certificates-2019.3.9-hecc5488_0
  certifi            conda-forge/win-64::certifi-2019.3.9-py37_0
  openssl            conda-forge/win-64::openssl-1.1.1b-hfa6e2cd_2
  pip                conda-forge/win-64::pip-19.1-py37_0
  python             conda-forge/win-64::python-3.7.3-hb12ca83_0
  setuptools         conda-forge/win-64::setuptools-41.0.1-py37_0
  sqlite             conda-forge/win-64::sqlite-3.26.0-hfa6e2cd_1001
  vc                 conda-forge/win-64::vc-14-0
  vs2015_runtime     conda-forge/win-64::vs2015_runtime-14.0.25420-0
  wheel              conda-forge/win-64::wheel-0.33.1-py37_0
  wincertstore       conda-forge/win-64::wincertstore-0.2-py37_1002


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate no-nbstripout
#
# To deactivate an active environment, use
#
#     $ conda deactivate

λ conda activate no-nbstripout

(no-nbstripout) λ conda list | grep nbstripout
# packages in environment at C:\Users\Centr\Anaconda3\envs\no-nbstripout:

(no-nbstripout) λ conda list # sanity check
# packages in environment at C:\Users\Centr\Anaconda3\envs\no-nbstripout:
#
# Name                    Version                   Build  Channel
ca-certificates           2019.3.9             hecc5488_0    conda-forge
certifi                   2019.3.9                 py37_0    conda-forge
openssl                   1.1.1b               hfa6e2cd_2    conda-forge
pip                       19.1                     py37_0    conda-forge
python                    3.7.3                hb12ca83_0    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
sqlite                    3.26.0            hfa6e2cd_1001    conda-forge
vc                        14                            0    conda-forge
vs2015_runtime            14.0.25420                    0    conda-forge
wheel                     0.33.1                   py37_0    conda-forge
wincertstore              0.2                   py37_1002    conda-forge

(no-nbstripout) λ nbstripout --is-installed && echo Success!
Success!

(no-nbstripout) λ where nbstripout
C:\Users\Centr\Anaconda3\Scripts\nbstripout.exe
```

Reply:
> We could make the hook try and specifically call the nbstripout that is installed? As in, instead of it calling the nbstripout that is within the PATH have it call the one that is referenced in the git nbstripout installation.

This could work and might be cleaner, actually. But I'm not sure I see the implementation yet. How in particular would you do this?

Reply:
> protects the case where people are accidentally committing to pymedphys without their env activated (not the hardest mistake to make).

This is actually one reason why having `.vscode/settings.json` in the `pymedphys` repo has been a bugbear for me. I really just want to add it to `.gitignore`, because I'm forever having to manually set my python interpreter to `pmp` in VS Code, which auto adds a `pythonPath` setting to `settings.json`. I'm forever unstaging or stashing that file...

Reply:
> ...to make that work we'll need to remove nbstripout from the bootstrap code. Otherwise it will unknowingly replace the base version with an nbstripout installed within the env.

I don't think anything will be replaced? There will just be multiple installs of `nbstripout` (read: multiple entires in `PATH`), which will 99.99999% of the time be fine?

If we have contributors savvy enough to mess with `nbstripout` settings such that multiple installs would be a problem, I think we just need to trust taht they know what they're doing...

Reply:
I didn't realise that's how base worked. Awesome. So have nbstripout in base. Easy fix. :) Sorry I made that so difficult.